tests/core/basic20/task.yaml

```diff
@@ -26,20 +26,22 @@ execute: |
     echo "Ensure passwd/group is available for snaps"
     test-snapd-sh-core20.sh -c 'cat /var/lib/extrausers/passwd' | MATCH test
 
-    #shellcheck source=tests/lib/names.sh
-    . "$TESTSLIB"/names.sh
+    if [[ "$SPREAD_SYSTEM" != ubuntu-core-20-arm-* ]]; then
+        #shellcheck source=tests/lib/names.sh
+        . "$TESTSLIB"/names.sh
 
-    echo "Ensure extracted kernel.efi exists"
-    test -e /boot/grub/"$kernel_name"*/kernel.efi
-    test -e /boot/grub/kernel.efi
+        echo "Ensure extracted kernel.efi exists"
+        test -e /boot/grub/"$kernel_name"*/kernel.efi
+        test -e /boot/grub/kernel.efi
+
+        echo "Ensure we are using managed boot assets"
+        MATCH '# Snapd-Boot-Config-Edition: [0-9]+' < /boot/grub/grub.cfg
+        MATCH '# Snapd-Boot-Config-Edition: [0-9]+' < /run/mnt/ubuntu-seed/EFI/ubuntu/grub.cfg
```

Comment:
I think this check would be the only thing that we check underneath the `if` that @mvo5 proposed, i.e.

```bash
#shellcheck source=tests/lib/names.sh
. "$TESTSLIB"/names.sh

if [[ "$SPREAD_SYSTEM" = ubuntu-core-20-64 ]]; then
        echo "Ensure extracted kernel.efi exists"
        test -e /boot/grub/"$kernel_name"*/kernel.efi

        echo "Ensure kernel.efi is a symlink"
        test -L /boot/grub/kernel.efi

        echo "Ensure we are using managed boot assets"
        MATCH '# Snapd-Boot-Config-Edition: [0-9]+' < /boot/grub/grub.cfg
        MATCH '# Snapd-Boot-Config-Edition: [0-9]+' < /run/mnt/ubuntu-seed/EFI/ubuntu/grub.cfg
else
        echo "Ensure extracted {kernel,initrd}.img exists
        test -e "/run/mnt/ubuntu-seed/systems/*/kernel/"$kernel_name"*/kernel.img"
        test -e "/run/mnt/ubuntu-seed/systems/*/kernel/"$kernel_name"*/initrd.img"
fi
```